x/feegrant/types/expiration.go

```diff
@@ -0,0 +1,138 @@
+package types
+
+import (
+	"time"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ExpiresAtTime creates an expiration at the given time
+func ExpiresAtTime(t time.Time) ExpiresAt {
+	return ExpiresAt{
+		Sum: &ExpiresAt_Time{
+			Time: &t,
+		},
+	}
+}
+
+// ExpiresAtHeight creates an expiration at the given height
+func ExpiresAtHeight(h int64) ExpiresAt {
+	return ExpiresAt{
+		&ExpiresAt_Height{
+			Height: h,
+		},
+	}
+}
+
+// ValidateBasic performs basic sanity checks.
+// Note that empty expiration is allowed
+func (e ExpiresAt) ValidateBasic() error {
+	if e.HasDefinedTime() && e.GetHeight() != 0 {
+		return sdkerrors.Wrap(ErrInvalidDuration, "both time and height are set")
+	}
+	if e.GetHeight() < 0 {
+		return sdkerrors.Wrap(ErrInvalidDuration, "negative height")
+	}
+	return nil
+}
+
+// Undefined returns true for an uninitialized struct
+func (e ExpiresAt) Undefined() bool {
+	return (e.GetTime() == nil || e.GetTime().Unix() <= 0) && e.GetHeight() == 0
+}
+
+// HasDefinedTime returns true if `ExpiresAt` has valid time
+func (e ExpiresAt) HasDefinedTime() bool {
```

Comment:
We have `Undefined` method, so for symmetry , how about  calling this method `IsDefined`?